packages/constellation/src/components/Compositions/Table/Cells/ButtonCell/ButtonCell.types.ts

```diff
@@ -0,0 +1,32 @@
+import { FC } from 'react'
+
+import { BaseCellProps } from '../..'
+
+interface ButtonProps {
+  /**
+   * The function that will be run on button click.
+   */
+  onClick: () => void
+  /**
+   * The text content of the button.
+   */
+  text: string
+  /**
+   * The stylistic variant to use when rendering the button.
+   */
+  variant?: 'primary' | 'secondary' | 'minimal'
+}
```

Comment:
It feels a little strange to maintain a separate set of Button props here, which are ultimately used for a Button component. Same concern with the other extended cell types like CheckboxCell, TagCell, etc. I think you could run into the situation where these props fall out of sync and a prop on the parent component causes an error with the child Button component.